main.tf

```diff
@@ -14,15 +14,42 @@
  * limitations under the License.
  */
 
+provider "google" {
```

Comment:
Providers shouldn't be defined in modules. Instead you can use a constraint - https://github.com/terraform-google-modules/terraform-google-network/blob/master/versions.tf#L17